src/CMakeLists.txt

```diff
@@ -10,7 +10,21 @@ include_directories(${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
 
 add_subdirectory(crypto)
 add_subdirectory(univalue)
+
+add_custom_target(generate_build_info
```

Comment:
Hmm, I don't think this will re-run on each `make`, so it won't pick up new git checkouts will it?

In other words: in master, switching git branches, or creating a new commit, will update build.h automatically on the next `make`. I don't think that's the case here?

Reply:
Huh, ok, yep. I didn't get that _at all_ from reading the code, but sure enough that's how it works :)